src/components/source_editor/Pyodide.jsx

```diff
@@ -0,0 +1,136 @@
+// eslint-ignore-file
+import CodeMirror from "@exabyte-io/cove.js/dist/other/codemirror/CodeMirror";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import Accordion from "@mui/material/Accordion";
+import AccordionDetails from "@mui/material/AccordionDetails";
+import AccordionSummary from "@mui/material/AccordionSummary";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import setClass from "classnames";
+import PropTypes from "prop-types";
+// eslint-disable-next-line no-unused-vars
+// import { loadPyodide } from "pyodide";
+import React from "react";
+
+class Pyodide extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            pythonCode: `
+import micropip
+print(globals())
+
+global my_dict
+my_dict = {"name": "pyodide"}
+
+def func():
+    return my_dict
+
+
+func()
+            #micropip.install("https://files.mat3ra.com:44318/web/pyodide/pymatgen-2023.9.10-py3-none-any.whl")
+            `,
+            result: "",
+        };
+        this.handleCodeChange = this.handleCodeChange.bind(this);
+        this.runPythonCode = this.runPythonCode.bind(this);
+    }
+
+    async componentDidMount() {
+        this.setState({ isLoading: true });
+        this.pyodide = await window.loadPyodide();
+        await this.pyodide.loadPackage("numpy");
+        await this.pyodide.loadPackage("micropip");
+        const micropip = this.pyodide.pyimport("micropip");
+        await micropip.install("ase");
+        this.setState({ isLoading: false });
+    }
+
+    handleCodeChange(newContent) {
+        this.setState({ pythonCode: newContent });
+    }
+
+    async runPythonCode() {
+        // eslint-disable-next-line no-unused-vars
+        const { pythonCode } = this.state;
+        // eslint-disable-next-line no-undef,react/destructuring-assignment
+        const convertedData = this.pyodide.toPy({ material: this.props.material.toJSON() });
+        try {
+            const result = await this.pyodide.runPythonAsync(pythonCode, {
+                globals: convertedData,
+            });
+            const dict = await this.pyodide.globals.toJs();
+            console.log("dict:", dict);
+            console.log("RESULT:", result);
```

Comment:
@VsevolodX we need `result.toJs()` to convert python dictionary into JS map. Then we can use `.get(key)` to extract values. Note `result` holds only functional return from python code.

Reply:
Basically this means we can only get the result of one run of code (the return of the last called function to be exact). A bit limiting, but in this particular usage, it's sufficient.